File: tests/test_formato.mjs
// Corredor sobre el fixture compartido tests/formato_casos.json.
//
// Verifica que plomada/static/formato.js produce lo mismo que quedo
// registrado en el fixture, generado leyendo plomada/data.py (la
// referencia). El otro lado del mismo fixture es tests/test_formato.py.
//
// Corre con: node tests/test_formato.mjs

import { readFileSync } from 'node:fs';
import { fileURLToPath } from 'node:url';
import path from 'node:path';
import { plata, pct, entero, titulo, slug, sinTildes } from '../plomada/static/formato.js';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const casos = JSON.parse(readFileSync(path.join(__dirname, 'formato_casos.json'), 'utf-8'));

// sin_tildes en data.py se llama sinTildes en formato.js: unico nombre que
// cambia entre los dos lados del fixture (los demas son iguales en Python
// y en JS).
const FNS = { plata, pct, entero, titulo, slug, sin_tildes: sinTildes };

let fallos = 0;
casos.forEach((c, i) => {
  const fn = FNS[c.fn];
  let obtenido;
  try {
    obtenido = fn(...c.args);
  } catch (e) {
    obtenido = `<excepcion: ${e.message}>`;
  }
  if (obtenido !== c.esperado) {
    fallos++;
    console.log(`  FALLO caso ${i}: ${c.fn}(${JSON.stringify(c.args)}) = ` +
      `${JSON.stringify(obtenido)}, esperaba ${JSON.stringify(c.esperado)}`);
  }
});

if (fallos) {
  console.log(`\n${fallos} de ${casos.length} casos fallaron`);
  process.exit(1);
}
console.log(`${casos.length} casos OK (node / formato.js)`);

File: tests/test_formato.py
"""Corredor sin pytest (no esta instalado en este equipo) sobre el fixture
compartido tests/formato_casos.json.

Verifica que plomada/data.py — el comportamiento de referencia — sigue
produciendo lo mismo que quedo registrado en el fixture. El otro lado del
mismo fixture es test_formato.mjs, que corre las mismas entradas contra
plomada/static/formato.js. Si un caso falla en uno pero no en el otro, las
dos implementaciones divergieron.

Corre con: python3 tests/test_formato.py
"""
import json
import sys
from pathlib import Path

RAIZ = Path(__file__).parent.parent
sys.path.insert(0, str(RAIZ / "plomada"))
import data as D  # noqa: E402

CASOS = json.loads((RAIZ / "tests" / "formato_casos.json").read_text(encoding="utf-8"))


def main():
    fallos = 0
    for i, c in enumerate(CASOS):
        fn = getattr(D, c["fn"])
        try:
            obtenido = fn(*c["args"])
        except Exception as e:  # una excepcion tambien es un fallo, no un crash del corredor
            obtenido = f"<excepcion: {e!r}>"
        if obtenido != c["esperado"]:
            fallos += 1
            print(f"  FALLO caso {i}: {c['fn']}({c['args']!r}) = {obtenido!r}, "
                  f"esperaba {c['esperado']!r}")
    if fallos:
        print(f"\n{fallos} de {len(CASOS)} casos fallaron")
        sys.exit(1)
    print(f"{len(CASOS)} casos OK (python / data.py)")


if __name__ == "__main__":
    main()
